pkg/scheduler/context.go

```diff
@@ -201,31 +201,106 @@ func (cc *ClusterContext) processRMConfigUpdateEvent(event *rmevent.RMConfigUpda
 	configs.ConfigContext.Set(cc.policyGroup, conf)
 }
 
-// Main update processing: the RM passes a large multi part update which needs to be unravelled.
-// Order of following operations is fixed, don't change unless carefully thought through.
-// 1) applications
-// 2) allocations on existing applications
-// 3) nodes
-// Updating allocations on existing applications requires the application to exist.
-// Node updates include recovered nodes which are linked to applications that must exist.
-func (cc *ClusterContext) processRMUpdateEvent(event *rmevent.RMUpdateRequestEvent) {
+func (cc *ClusterContext) handleRMUpdateNodeEvent(event *rmevent.RMUpdateNodeEvent) {
 	request := event.Request
-	// 1) Add / remove app requested by RM.
-	cc.processApplications(request)
-	// 2) Add new request, release allocation, cancel request
-	cc.processAllocations(request)
-	// 3) Add / remove / update Nodes
 	cc.processNodes(request)
 }
 
-func (cc *ClusterContext) processNodes(request *si.UpdateRequest) {
-	// 3) Add / remove / update Nodes
-	// Process add node
-	if len(request.NewSchedulableNodes) > 0 {
-		cc.addNodes(request)
-	}
-	if len(request.UpdatedNodes) > 0 {
-		cc.updateNodes(request)
+func (cc *ClusterContext) processNodes(request *si.NodeRequest) {
+	// 1) Add / remove / update Nodes
+	if len(request.GetNodes()) > 0 {
+		acceptedNodes := make([]*si.AcceptedNode, 0)
+		rejectedNodes := make([]*si.RejectedNode, 0)
+		for _, nodeInfo := range request.GetNodes() {
+			switch nodeInfo.Action {
+			case si.NodeInfo_CREATE:
+				sn := objects.NewNode(nodeInfo)
```

Comment:
Can we move this create into its own method? Similar to the way `updateNode` is called for all update cases? Makes the code more readable:
```
// addNode adds a node to the cluster returns error if the action fails or nil on success
func (cc *ClusterContext) addNode(nodeInfo *si.NodeInfo) error {
}
```
If the response is a an error the error message can be used in the reject state otherwise it is an accepted node.